lib/galaxy/jobs/__init__.py

```diff
@@ -1770,10 +1777,14 @@ def reclaim_ownership( self ):
     def user_system_pwent( self ):
         if self.__user_system_pwent is None:
             job = self.get_job()
-            try:
-                self.__user_system_pwent = pwd.getpwnam( job.user.email.split('@')[0] )
-            except:
-                pass
+
+            # try to get user from email address/username
+            for cand in [job.user.email.split('@')[0], job.user.username]:
```

Comment:
Sounds possible. 

The question is if this is necessary, since testing both adds only a little computation (I guess its the same as the if-statement that would be necessary when we make this configurable). 

For me both possibilities would be fine. 

If we leave it as is I could imagine in case that both ways to determine a user are successful a test checking if they yield the same user would be a good idea. 

Otherwise: Can somebody point me to examples showing me how to access configuration variables?